awx/api/views.py

```diff
@@ -729,6 +731,51 @@ def delete(self, request):
         return Response(status=status.HTTP_204_NO_CONTENT)
 
 
+class UserSessionView(APIView):
```

Comment:
I don't think this class is needed at all, at least not in its current form.

On down the PR you create a new table for Session membership but in reality we can probably just override the DRF provided auth view that I mentioned above and hook into its POST method to handle our built-in authentication logic.

`AUTH_TOKEN_PER_USER` should probably change (or at least, a new one should be created and this one should exist for backwards compatibility.

All we really need to do is go back and look at the session table and force expire ones when it crosses the per-user threshold.

Another *really* important setting to rethink is `AUTH_TOKEN_EXPIRATION`. This will just work differently under oauth tokens. Django has another setting which controls session duration: https://docs.djangoproject.com/en/1.11/ref/settings/#session-cookie-age

Note that session cookies can be set to expire at browser close also... this is a very common use case.

Reply:
The purpose of introducing the new table is performance. The `Session` table comes with only 3 fields, `session_key`, `session_value` and `expiration_date`. The user association information is encoded inside `session_value`, meaning if we want to find session that are crossing threshold of *a particular user*, we will have to fetch *all* unexired sessions, and for each one of them decode `session_value` field and inspect decoded JSON to find if it belongs to that user. If we don't want to introduce an auxiliary table purely for speed up, maybe we can utilize cache?

`AUTH_TOKEN_EXPIRATION` is a good catch, and session duration is surly better off being configurable. Will take them in list, along with other items like periodic session clear.